src/Illuminate/Auth/Access/Gate.php

```diff
@@ -708,6 +713,32 @@ protected function guessPolicyName($class)
         }) ?: [$classDirname.'\\Policies\\'.class_basename($class).'Policy']);
     }
 
+    /**
+     * Try to resolve the policy via the UsePolicy attribute.
+     *
+     * @param  class-string  $class
+     * @return mixed
+     */
+    protected function tryResolvePolicyViaUsePolicyAttribute($class)
+    {
+        try {
+            $attributes = (new \ReflectionClass($class))
+                ->getAttributes(UsePolicy::class);
+        } catch (Exception) {
+            return null;
+        }
+
+        if ($attributes !== []) {
+            $usePolicy = $attributes[0]->newInstance();
+
+            $policy = new $usePolicy->policyClass;
+
+            return $policy;
```

Comment:
This should resolve the policy via the container.

```suggestion
            return $this->resolvePolicy($usePolicy->policyClass);
```

Reply:
Good point! Thanks